src/interrupt.rs

```diff
@@ -3,9 +3,27 @@
 //! For the most part, [crate::interrupt::free] is what you want:
 //!
 //! ```
-//! atmega32u4::interrupt::free(|cs| {
+//! avr_device::interrupt::free(|cs| {
 //!     // Interrupts are disabled here
-//! })
+//! });
+//! ```
+//!
+//! To access shared state, Mutex can be used:
+//!
+//! ```
+//! use avr_device::interrupt::Mutex;
+//! use core::cell::Cell;
+//!
+//! static MYGLOBAL: Mutex<Cell<u16>> = Mutex::new(Cell::new(0));
```

Comment:
I think it would be valuable to point out that `Cell` should be used for `Copy` types and `RefCell` when a reference to the data inside the mutex is needed at some point.